Dockerfile

```diff
@@ -1,42 +1,82 @@
-FROM golang:1.17-alpine as cloudreve_builder
+# 多阶段构建: 编译发布物
+# 注意: 该阶段除最终发布物外不会影响最终镜像体积, 删除缓存、减少包安装等无意义
+FROM golang:1-alpine AS builder
 
+# 安装基本编译依赖
+RUN set -ex \
+    && apk add build-base bash git yarn zip \
+    && git clone --recursive https://github.com/cloudreve/Cloudreve.git /source
 
-# install dependencies and build tools
-RUN apk update && apk add --no-cache wget curl git yarn build-base gcc abuild binutils binutils-doc gcc-doc zip
+# 编译前端代码
+WORKDIR /source/assets
 
-WORKDIR /cloudreve_builder
-RUN git clone --recurse-submodules https://github.com/cloudreve/Cloudreve.git
+# 允许通过 `docker build --build-args YARN_REGISTRY=https://registry.npmmirror.com` 覆盖默认仓库地址
+#
+# 常用的仓库地址:
+#   npm -----  https://registry.npmjs.org
+#   cnpm ----  http://r.cnpmjs.org
+#   taobao --  https://registry.npmmirror.com
+#   nj ------  https://registry.nodejitsu.com
+#   rednpm --  http://registry.mirror.cqupt.edu.cn
+#   skimdb --  https://skimdb.npmjs.com/registry
+#   yarn ----  https://registry.yarnpkg.com
+ARG YARN_REGISTRY=https://registry.yarnpkg.com
```

Comment:
normally, I think this is no need, we publish this image to docker hub after actions build done, you can specify these in Cloudreve docs, you can think this is a docker image that we deliver to user directly, not help them to build, the tutorial work should be put in docs.